langchain4j-core/src/main/java/dev/langchain4j/data/image/Image.java

```diff
@@ -0,0 +1,36 @@
+package dev.langchain4j.data.image;
+
+public class Image {
+
+    private String url;
```

Comment:
What if an image is represented as a Base64 encoded string?
For example, in Gemini, an image is either passed as a Base64 encoded string or as a URI to a google cloud storage bucket.
I fear this Image class is not generic enough to work with other models that support images.

Reply:
My intention was to keep `Image` ready-to-use for the client as much as possible. This means if Base64 is configured, then the image(s) download happens under the hood. In the end the client gets the URL to local file here. Check #298, there is an inte test too.

Reply:
I agree with @glaforge, we should add base64 content field to `Image` to make it more versatile. We should also not force user to use our built-in persisting logic, it should be an optional "nice to have".

Reply:
I'm currently coding this part, and indeed I agree too with @glaforge - it feels strange to write the image locally, for me it should be passed in the base64 field instead.

Reply:
Maybe also adding the mime type, to accomodate PNG, JPEG, etc.

Reply:
I'm fine with adding base64 field to `Image`, sure.

Reply:
I called it `base64Data` as it's the name used in Azure OpenAI and that allows to have a `base64MimeType` in the future to do like @glaforge suggested.

Reply:
I think we're good for the moment, so marking this as resolved